app/static/src/app/store/appState/actions.ts

```diff
@@ -29,5 +44,28 @@ export const appStateActions: ActionTree<AppState, AppState> = {
                 await dispatch(`model/${ModelAction.DefaultModel}`);
             }
         }
+    },
+
+    async [AppStateAction.QueueStateUpload](context) {
+        const { state, commit } = context;
+        const isFitting = () => { return (state.appType === AppType.Fit) && ((state as FitState).modelFit.fitting); };
+
+        // Do not queue uploads while fitting is true - we'll upload when fit finishes
+        if (!isFitting()) {
+            // remove any existing queued upload, as this request should supersede it
+            commit(AppStateMutation.ClearQueuedStateUpload);
+
+            const queuedId: number = window.setInterval(() => {
+                // wait for any ongoing uploads to finish before starting a new one
```

Comment:
Does this mean that there will be a 2s gap between the _end_ of one upload and the start of the next, or the _start_ of one upload and the start of the next?

Reply:
It means that any mutation that happen within 2s of eachother will get queued (waiting uploads get superseded by a new queued upload if a mutation come in in the meantime). So the worst luck scenario would be, as you say that an upload is commenced, and then a new mutation immediately comes in and starts two seconds later (assuming the first upload has finished by then - if not it will check every 2 seconds until it has).